File: ReverseString/ReverseString.py
class Solution(object):
    def reverseString(self, s):
        """
        :type s: str
        :rtype: str
        """
        listS = list(s)
        first = 0
        last = len(listS) - 1
        while first < last:
        	tmp = listS[first]
        	listS[first] = listS[last]
        	listS[last] = tmp
        	first += 1
        	last -= 1
        return "".join(listS)

if __name__ == '__main__':
	A = Solution()
	s = A.reverseString("123")
	print s

File: ReverseString/ReverseString.cpp
#include <string>

using namespace std;

class Solution {
public:
    string reverseString(string s) {
		if (s.size() == 0)
			return s;
       	string::size_type first = 0;
    	string::size_type last = s.size() - 1;
    	while(first < last) {
    		char tmp = s[first];
    		s[first] = s[last];
    		s[last] = tmp;
    		first++;
    		last--;
    	}
    	return s;
    }
};